### PR TITLE
More robust check for whether a filepath is remote or not

### PR DESCRIPTION
**Pull request by katrogan**
Signed-off-by: Katrina Rogan <katroganGH@gmail.com>

# TL;DR
This updates the check used for [FlyteFile](https://github.com/flyteorg/flytekit/blob/532ff368aa51319b77b0430c90f37f15f3d6afe5/flytekit/types/file/file.py#L333)s and [FlyteDirectory](https://github.com/flyteorg/flytekit/blob/532ff368aa51319b77b0430c90f37f15f3d6afe5/flytekit/types/directory/types.py#L272)s to see if a user-provided path is remote or not.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Discovered bug while testing out mpi plugin.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1518

## Follow-up issue
_NA_

### CODE REVIEW

File: flytekit/core/data_persistence.py [hunk 1]
```diff
@@ -258,7 +304,10 @@ def is_remote(path: Union[str, os.PathLike]) -> bool:
         """
         Deprecated. Lets find a replacement
         """
-        return not (path.startswith("/") or path.startswith("file://"))  # type: ignore
+        protocol, _ = split_protocol(path)
```
Comment (kumare3) [inline flytekit/core/data_persistence.py:307]: should we replace here too - https://github.com/flyteorg/flytekit/blob/master/flytekit/extras/persistence/s3_awscli.py#L85?

Comment (kumare3) [reply flytekit/core/data_persistence.py:307]: i guess not needed

**Review by kumare3 — APPROVED** [review]

File: flytekit/core/data_persistence.py (hunk 1 above)
Comment (katrogan) [reply flytekit/core/data_persistence.py:307]: yeah that should always be a string, although i guess we could move to using split_protocol there
